tools/run_all_tests.py

```diff
@@ -307,9 +307,54 @@ def test_all(
     return num_failing + num_timing_out == 0
 
 
+def warm_start_app():
+    """Warm start the app to compile shaders before running the tests."""
+    print("[INFO] Warm starting the simulation app before running tests.")
+    before = time.time()
+    # headless experience
+    warm_start_output = subprocess.run(
+        [
+            sys.executable,
+            "-c",
+            (
+                "from omni.isaac.lab.app import AppLauncher; app_launcher = AppLauncher(headless=True);"
+                " app_launcher.app.close()"
+            ),
+        ],
+        capture_output=True,
+    )
+    if len(warm_start_output.stderr) > 0:
+        logging.error(f"Error warm starting the app: {str(warm_start_output.stderr)}")
+        exit(1)
+
+    # headless experience with rendering
```

Comment:
Just doing the one with rendering should take care of the first case too? I don't think you need separate caching for the two cases? Or do you?